PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/messager/IntellijAzureMessager.java

```diff
@@ -6,65 +6,84 @@
-import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
 
 import javax.annotation.Nonnull;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
 public class IntellijAzureMessager implements IAzureMessager {
     private static final String NOTIFICATION_GROUP_ID = "Azure Plugin";
     private static final String DEFAULT_MESSAGE_TITLE = "Azure";
 
-    private void showNotification(@Nonnull String title, @Nonnull String message, NotificationType type) {
-        Notifications.Bus.notify(new Notification(NOTIFICATION_GROUP_ID, title, message, type));
+    private Notification createNotification(@Nonnull String title, @Nonnull String message, NotificationType type) {
+        return new Notification(NOTIFICATION_GROUP_ID, title, message, type, new NotificationListener.UrlOpeningListener(true));
     }
 
-    private String getTitle(String... title) {
-        if (ArrayUtils.isEmpty(title)) {
+    private String getTitle(String title) {
+        if (StringUtils.isEmpty(title)) {
             return DEFAULT_MESSAGE_TITLE;
         }
-        return title[0];
+        return title;
     }
 
-    public boolean confirm(@Nonnull String message, String title) {
-        return MessageDialogBuilder.yesNo(getTitle(title), message).guessWindowAndAsk();
-    }
-
-    public void alert(@Nonnull String message, String title) {
-        MessageDialogBuilder.okCancel(getTitle(title), message).guessWindowAndAsk();
-    }
-
-    public void success(@Nonnull String message, String title) {
-        this.showNotification(getTitle(title), message, NotificationType.INFORMATION);
-    }
-
-    public void info(@Nonnull String message, String title) {
-        this.showNotification(getTitle(title), message, NotificationType.INFORMATION);
-    }
-
-    public void warning(@Nonnull String message, String title) {
-        this.showNotification(getTitle(title), message, NotificationType.WARNING);
-    }
-
-    public void error(@Nonnull String message, String title) {
-        this.showNotification(getTitle(title), message, NotificationType.ERROR);
-    }
-
-    public void error(@Nonnull Throwable throwable, String title) {
-        IntelliJAzureExceptionHandler.getInstance().handleException(throwable);
+    public String value(String val) {
+        return val;
     }
 
-    public void error(@Nonnull Throwable throwable, @Nonnull String message, String title) {
-        final AzureToolkitRuntimeException wrapped = new AzureToolkitRuntimeException(message, throwable);
-        IntelliJAzureExceptionHandler.getInstance().handleException(wrapped);
+    @Override
+    public boolean show(IAzureMessage message) {
+        final NotificationType type;
+        switch (message.getType()) {
+            case ALERT:
+                return MessageDialogBuilder.okCancel(getTitle(message.getTitle()), message.getMessage()).guessWindowAndAsk();
```

Comment:
Not sure whether we should pop up dialogs for `alert`, as for error we will handle it with `ExceptionHandler`, where only forehead operations will show error dialog and break user's operations. But here we will always break users' operation for `ALERT`